### PR TITLE
Fix flow and check more

### PR DESCRIPTION
**Pull request by rgbkrk**
This likely won't pass because of #2360, namely that packages aren't resolving properly from applications that have their package name as `@nteract`...

### CODE REVIEW

**Comment by betatim** [comment]
What do we conclude from the fact that it did pass?

**Comment by rgbkrk** [comment]
I'm thinking that _somehow_ locally my setup wasn't fully bootstrapped (each of our packages weren't linked in these apps for some reason).

**Comment by rgbkrk** [comment]
I was able to fix it locally (and quite sadly) by doing the `git clean -xfd && npm i` trick.

File: applications/notebook-on-next/pages/_document.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
+// @flow
 import React from "react";
 import Document, { Head, Main, NextScript } from "next/document";
 
 export default class MyDocument extends Document {
-  static getInitialProps({ renderPage }) {
+  static getInitialProps({ renderPage }: { renderPage: Function }) {
```
Comment (betatim) [inline applications/notebook-on-next/pages/_document.js:6]: What does this do? Not sure I've seen this notation before. `renderPage` is of type `Function` but `renderPage` is also of type hash with key renderPage ?

Comment (rgbkrk) [reply applications/notebook-on-next/pages/_document.js:6]: I'll expand this a bit locally to explain --

```js
// getInitialProps is a function unique to next.js, not normal to React
static getInitialProps
// Apparently our flow setup doesn't have this definition already situated
// The first (and only?) argument is an object
// which has a property called renderPage
({ renderPage }:
// Everything after the ":" is the actual type definition for this
// so we're saying the type of this argument is an object with property renderPage,
// which is a Function (which I could have typed out the type of more explicitly --
// I aimed for passing for now)
{ renderPage: Function })
```

Comment (betatim) [reply applications/notebook-on-next/pages/_document.js:6]: ah! I had forgotten about the unpacking via `{ keyHere }`! Makes sense.

**Comment by betatim** [comment]
merges in return for eduction. I think this is what they call a win-win :)

**Pull request merged**